### PR TITLE
register cloud events metrics.

### PR DESCRIPTION
**Pull request by morvencao**
ref: https://issues.redhat.com/browse/ACM-14351

### CODE REVIEW

File: pkg/client/cloudevents/source_client.go [hunk 1]
```diff
@@ -42,7 +43,7 @@ func NewSourceClient(sourceOptions *ceoptions.CloudEventsSourceOptions, resource
 	}
 
 	// register resource resync metrics for cloud event source client
-	cegeneric.RegisterResourceResyncMetrics()
+	cegeneric.RegisterCloudEventsMetrics(prometheus.DefaultRegisterer)
```
[the remark this answers is not in the document]
Comment (morvencao) [reply pkg/client/cloudevents/source_client.go:46]: `legacyregistry` is used by maestro-agent with library-go to register metrics, and `prometheus.DefaultRegisterer` is typically used with the Prometheus SDK to register metrics.

**Comment by clyang82** [comment]
Do we have e2e test for metrics?

**Comment by morvencao** [comment]
@clyang82 added simple integration test as UT cover the metrics in sdk-go repo.

**Review by clyang82 — APPROVED** [review]
LGMT

**Pull request merged**